src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -505,6 +503,9 @@ SDCPUThread::onExecutionCompleted(weak_ptr<InflightInst> inst, Fault fault)
         return;
     }
 
+    // if (!inst->readPredicate()) // TODO ensure this is enforced.
```

Comment:
If this isn't needed, it should be deleted.

Reply:
Oh that was just a commented line I added because I noticed either O3 or TimingSimple doing that so I was like maybe I should take a note

Reply:
Make sense.